### PR TITLE
External file for rule confidence

### PR DESCRIPTION
**Pull request by danielnaber**
Load rule confidence from am external file. This is limited to one confidence per match/rule, even though the match might have more than one suggestion. They will all end up with the same confidence and the client needs to interpret that...

### CODE REVIEW

File: languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/CompoundAcceptor.java [hunk 1]
```diff
@@ -1099,7 +1099,7 @@ public class CompoundAcceptor {
     }
   }
 
-  private DutchTagger dutchTagger = DutchTagger.INSTANCE;
+  private final DutchTagger dutchTagger = DutchTagger.INSTANCE;
```
Comment (danielnaber) [inline languagetool-language-modules/nl/src/main/java/org/languagetool/rules/nl/CompoundAcceptor.java:1102]: sorry, unrelated commit

File: languagetool-server/src/main/java/org/languagetool/server/HTTPServerConfig.java [hunk 2]
```diff
@@ -1503,4 +1512,10 @@ public void setDbMaxConnections(int dbMaxConnections) {
     this.dbMaxConnections = dbMaxConnections;
   }
 
+  /**
+   * @since 6.4
+   */
```
Comment (SteVio89) [inline languagetool-server/src/main/java/org/languagetool/server/HTTPServerConfig.java:1517]: Add @Nullable annotation

File: languagetool-server/src/main/java/org/languagetool/server/V2TextChecker.java [hunk 3]
```diff
@@ -39,9 +45,40 @@ class V2TextChecker extends TextChecker {
...
+  private void loadConfidenceMap() throws IOException {
+    if (config.getRuleIdToConfidenceFile() != null) {
+      logger.info("Loading confidence map for rules from " + ruleIdToConfidence);
+      List<String> lines = Files.readAllLines(Paths.get(config.getRuleIdToConfidenceFile().getAbsolutePath()), UTF_8);
+      for (String line : lines) {
+        if (line.startsWith("#")) {
+          continue;
+        }
+        String[] parts = line.split(",");
+        if (parts.length >= 2) {
```
Comment (SteVio89) [inline languagetool-server/src/main/java/org/languagetool/server/V2TextChecker.java:70]: Why `>= 2` and not `== 2`?

Comment (danielnaber) [reply languagetool-server/src/main/java/org/languagetool/server/V2TextChecker.java:70]: there might be more columns for better debugging, but we don't use them here  - I'll be adding that as a comment

**Review by SteVio89 — APPROVED** [review]
